### PR TITLE
fix(sync): If `baseVersionEtag` changed, reset frontend

### PR DESCRIPTION
**Pull request by mejo-**
### 📝 Summary

The PR builds on #5477 

* fix(sync): If `baseVersionEtag` changed, reset frontend
   `baseVersionEtag` changes when a new document session got initialized, e.g. after an old document session without session clients got cleaned up, or because the markdown file got changed via webdav.
   Detect this in the client and ask the user to reload the page for resetting the session.
* fix(Middleware): Response with 412 if `baseVersionEtag` doesn't match
* fix(DocumentStatus): Refactor and migrate to `NcNoteCard`

### Screenshot

| warning |
| --- |
| 2024-03-18T20:06:02,244342045+01:00 |

### 🚧 TODO

- [x] Testing

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests

### CODE REVIEW

File: cypress/e2e/sync.spec.js [hunk 1]
```diff
@@ -126,6 +126,25 @@ describe('Sync', () => {
 			.should('include', 'after the lost connection')
 	})
 
+	it.only('shows warning when document session got cleaned up', () => {
```
Comment (max-nextcloud) [inline cypress/e2e/sync.spec.js:129]:
```suggestion
	it('shows warning when document session got cleaned up', () => {
```

File: cypress/support/sessions.js [hunk 2]
```diff
@@ -50,16 +50,40 @@ Cypress.Commands.add('pushSteps', ({ connection, steps, version, awareness = ''
 		.then(response => response.data)
 })
 
+Cypress.Commands.add('failToPushSteps', ({ connection, steps, version, awareness = '' }) => {
+	return connection.push({ steps, version, awareness })
+		.then((response) => {
+			throw new Error('Expected request to fail - but it succeeded!')
+		})
+		.catch((err) => err.response)
```
Comment (max-nextcloud) [inline cypress/support/sessions.js:58]: I think the `throw` in `then` will be caught in the catch and cause a `undefined` to be returned as `err.response` is undefined for the exception thrown.
You probably want to handle both cases in one [`then(onFullfilled, onRejected)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#syntax) call:
```suggestion
		.then(
			() => throw new Error('Expected request to fail - but it succeeded!'),
			err => err.response
		)
```

**Review by max-nextcloud — APPROVED** [review]
Looks good. Just some minor nitpicks.

File: cypress/support/sessions.js (hunk 2 above)
Comment (mejo-) [reply cypress/support/sessions.js]: Indeed, sounds logical. Changed now.

**Pull request merged**